source/dub/dub.d

```diff
@@ -556,17 +556,24 @@ class Dub {
 			return;
 		}
 
-		foreach (p; versions.byKey) {
+		import std.parallelism : TaskPool, totalCPUs;
+		// Avoid being blocked by a slow download (even on a single-threaded machines)
+        auto taskPool = new TaskPool(totalCPUs.clamp(3, 6));
+		foreach (p; taskPool.parallel(versions.byKey, 1)) {
```

Comment:
Have we tried to use `.byKey
array` here?

Reply:
No, not yet. We will want to do that in the final version.

During the experimenting, I reset the branch to start from your implementation. I noticed that, depending on the amount of logging I added, the behaviour was changing. Of course, as a race condition occurs, this is to be expected. I didn´t expect it would corrupt the array of dependencies (it once failed with an empty list of dependencies). The sanity check _after_ the fetch would fail (see the [commit](https://github.com/dlang/dub/pull/1704/commits/f95234b7e66f36085bb889b5a0a80c0d2ff39e9c#diff-fc1737ce75bc9102f5d5b54be258a268R159)). It might be the same cause (as in, the same read/write operation), but it might not be.